### PR TITLE
Stackdriver - Ability to configure credentials

### PR DESCRIPTION
**Pull request by saturnism**
Added an additional config so that the credentials can be configured programmatically.

### CODE REVIEW

File: implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverMeterRegistry.java [hunk 1]
```diff
@@ -85,8 +85,8 @@
     @Nullable
     private MetricServiceClient client;
 
-    public StackdriverMeterRegistry(StackdriverConfig config, Clock clock) {
-        this(config, clock, DEFAULT_THREAD_FACTORY, () -> MetricServiceSettings.newBuilder().build());
+    protected StackdriverMeterRegistry(StackdriverConfig config, Clock clock) {
```
Comment (shakuzen) [inline implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverMeterRegistry.java]: Why the change in access here? This will break code that was using this public constructor before.

Comment (shakuzen) [reply implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverMeterRegistry.java]: @saturnism I'd like to include this feature in the upcoming release. When you have time, let me know about the above question.

Comment (saturnism) [reply implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverMeterRegistry.java]: eak, not quite sure why it was changed ;) i've reverted it back. thnx for catching this!

**Comment by saturnism** [comment]
not sure why this is failing; built locally :/
